osquery/tables/networking/windows/arp_cache.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+*  Copyright (c) 2014-present, Facebook, Inc.
+*  All rights reserved.
+*
+*  This source code is licensed under the BSD-style license found in the
+*  LICENSE file in the root directory of this source tree. An additional grant
+*  of patent rights can be found in the PATENTS file in the same directory.
+*
+*/
+
+#include <boost/range/algorithm/find.hpp>
+#include <osquery/tables.h>
+#include <string>
+
+#include "boost/algorithm/string/replace.hpp"
+#include "osquery/core/conversions.h"
+#include "osquery/core/windows/wmi.h"
+#include "osquery/tables/networking/windows/interfaces.h"
+
+namespace osquery {
+namespace tables {
+
+const std::map<unsigned short, const std::string> mapOfAddressFamily = {
+    {2, "IPv4"}, {23, "IPv6"},
+};
+
+const std::map<unsigned char, const std::string> mapOfStore = {
+    {0, "Persistent"}, {1, "Active"},
+};
+
+const std::map<unsigned char, const std::string> mapOfState = {
+    {0, "Unreachable"},
+    {1, "Incomplete"},
+    {2, "Probe"},
+    {3, "Delay"},
+    {4, "Stale"},
+    {5, "Reachable"},
+    {6, "Permanent"},
+    {7, "TBD"},
+};
+
+QueryData genArpCache(QueryContext& context) {
+  Row r;
+  QueryData results;
+  QueryData interfaces = genInterfaceDetails(context);
+  WmiRequest wmiSystemReq("select * from MSFT_NetNeighbor",
+                          L"ROOT\\StandardCimv2");
+  std::vector<WmiResultItem>& wmiResults = wmiSystemReq.results();
+  std::map<long, std::string> mapOfInterfaces = {
+      {1, ""}, // loopback
+  };
+  unsigned short usiPlaceHolder;
+  unsigned char cPlaceHolder;
+  unsigned int uiPlaceHolder;
+  std::string strPlaceHolder;
+
+  for (auto& iface : interfaces) {
+    long interfaceIndex = atol(iface["interface"].c_str());
```

Comment:
Inside of `conversions.h` we have a `safestrtol` function that's a bit better suited for converting strings to longs. https://github.com/facebook/osquery/blob/257535e5a2fe22332955eec784419460e19fcdab/osquery/core/conversions.h#L136